### ISSUE TITLE
Change the way to store SymPy papers

### ISSUE THREAD

**Issue by nicoguaro** [sympy/sympy#11569]
I have been updating the [Sympy Papers wiki page](https://github.com/sympy/sympy/wiki/SymPy-Papers) regularly with new papers. But I think that is reaching a point where the list is getting too big to be useful.

Thus, I  thought that maybe we can store the bibliographic information (more than the info that we have right now) in  a different way, maybe a .bib file. And then, we can create a script to update this wiki page. 

What do you think about it?

**Comment by moorepants** [comment]
I started this when we started the recent SymPy paper:

https://www.zotero.org/groups/sympy

I'd be happy to add you to it. You can easily export to various file formats and reference lists with Zotero.

**Comment by nicoguaro** [comment]
I can certainly add the references there. That way more metadata of the documents can be recorded.

**Comment by moorepants** [comment]
If you make a zotero account and give me your username I'll add you.

**Comment by nicoguaro** [comment]
I am nicoguaro in Zotero.

**Comment by moorepants** [comment]
Ok, the invite has been sent.

**Comment by nicoguaro** [comment]
Should I create a new collection?

**Comment by moorepants** [comment]
You can just use the one that is there. I can rename it.

**Comment by nicoguaro** [comment]
Two more questions. Should I add keywords/tags? Should I attach PDF files?

I updated the database with a couple of references with testing purposes.

**Comment by moorepants** [comment]
Yes, add whatever you like. I have notes and tags on some of the existing entries. You can attach PDFs but there is a storage limit, so don't attach very large ones.

**Comment by nicoguaro** [comment]
I added all the documents in the section "Papers using SymPy". Should we also add the ones in the section "Papers mentioning SymPy"?

**Comment by moorepants** [comment]
Maybe those categories are worth having different tags to know whether sympy was used or just mentioned.

**Comment by nicoguaro** [comment]
It is done now.

**Comment by asmeurer** [comment]
That's awesome. I have an [alert](https://scholar.google.com/scholar_alerts?view_op=list_alerts&hl=en) set up for SymPy on Google Scholar and there are a couple of new papers about once a week. Most of them, maybe 75%, just mention sympy once in the paper but don't indicate how it was used (I'm guessing many just used SymPy as a calculator to get some result), but use it quite in-depth. 

Does zotero let you sort the papers by subject area, without manually tagging each paper with a subject?

**Comment by moorepants** [comment]
Zotero can sort by any of the fields it has and also by tags, keywords, etc. Unless the subject is in the meta data of the downloaded or manually entered citation, you won't be able to search by subject. What do you mean by "subject" anyways?

**Comment by asmeurer** [comment]
I'm mostly interested in high-level academic subject areas (biology, chemistry, etc.).

**Comment by moorepants** [comment]
I think your best bet is to look at what journal is it published in. There is no standard for tagging papers that fall into different science domains.

**Comment by asmeurer** [comment]
Ah, I was misunderstanding the UI. There are a bunch of tags on the paper pages, like https://www.zotero.org/groups/sympy/items/itemKey/KSDB3BPK, but those aren't actually the tags for that paper. I thought maybe a bunch of tags were added by zotero automatically, but I guess that's not the case.

**Comment by moorepants** [comment]
Some of the tags come from when you download a paper using the zotero browser button. It picks them up from the metadata provided by the journal or repository website. Otherwise you have to add them manually in the zotero interface. The browser extension and standalone zotero program provide a richer UI than the web app to do all of this.

**Comment by nicoguaro** [comment]
@asmeurer, as @moorepants mentioned, if the meta-data is associated to the DOI (or wherever you are retrieving the information) it will be added to the document entry. If you think that creating a rough categorization of the documents is worth it, then we can do it now that the list is "small". And then, we can suggest that any new entries should have those.

**Comment by nicoguaro** [comment]
The list of documents is 152 right now (and growing). Is it worth to add the list to the wiki?

**Comment by moorepants** [comment]
A regular export from zotero and pasting into the wiki would be fine I think.

**Comment by oscarbenjamin** [comment]
Looks like the list of papers hasn't been updated in a couple of years...

**Comment by nicoguaro** [comment]
@oscarbenjamim, I have updated the Zotero lis, but I have never updated the list in the website.

**Comment by moorepants** [comment]
Yes , papers have been added to zotero pretty regularly, thanks mostly to @nicoguaro. It is fairly easy to export a bibliography from zotero if you want to paste it into the wiki every so often. I can add more people to the zotero group too.

**Comment by nicoguaro** [comment]
I think that adding all papers is not that appealing. Is it an option to create a minisite on the website with an interactive view of the papers?

**Comment by asmeurer** [comment]
How hard would it be to add something like that?

**Comment by nicoguaro** [comment]
Sincerely, I have no idea. I will look around to get an idea.

**Comment by nicoguaro** [comment]
@asmeurer, I found [BibBase](https://bibbase.org/) and it seems pretty straightforward.

There are other options in the [Zotero website](https://www.zotero.org/support/dev/web_api/v3/start).

**Comment by asmeurer** [comment]
That looks nice. I don't have any strong opinions about it but maybe others will. A solution that uses Javascript to load everything so that it doesn't require a page rebuild to update seems the best. Do you think something like this should go on the website or in the docs? We obviously can't put anything like that on the wiki.

**Comment by nicoguaro** [comment]
I tried it creating an html using an ``<iframe>`` and the JS script and both worked.

Regarding where to place it, I am not sure. I would say we wait for some feedback from others' opinion (?).

**Comment by moorepants** [comment]
There is also this: https://github.com/urschrei/pyzotero

It could be used in the docs build to make an HTML page based on the zotero library.

**Comment by asmeurer** [comment]
It looks like that's just an API library. You would need to build a Sphinx extension that uses it to generate a page.

**Comment by moorepants** [comment]
> It looks like that's just an API library. You would need to build a Sphinx extension that uses it to generate a page.

You could probably get away with a script that generates html that could be included in a sphinx page with the raw directive.

**Comment by asmeurer** [comment]
BibBase looks like the nicest option. It lets you change the grouping, which is nice since there would be hundreds of entries, and download bibtex for each entry. It would be nice if it had a search filter as well.

Actually, it isn't clear to me if it grabs Zotero statically or dynamically. If it's statically, you would have to regenerate it to get the latest entries. I don't have a Zotero so may be one of you can check what it generates.

**Comment by moorepants** [comment]
https://bibbase.org/show?bib=https%3A%2F%2Fapi.zotero.org%2Fgroups%2F525293%2Fitems%3Fkey%3Dw47UuB5mugS0ACGIZckhpHFz%26format%3Dbibtex%26limit%3D100&msg=embed

**Comment by moorepants** [comment]
For some reason it isn't displaying all the papers.

**Comment by nicoguaro** [comment]
> Actually, it isn't clear to me if it grabs Zotero statically or dynamically. If it's statically, you would have to regenerate it to get the latest entries. I don't have a Zotero so may be one of you can check what it generates.

It does generates it dynamically, I just added entries to check it.

> For some reason it isn't displaying all the papers.

It seems that is related to the date format, but I am not certain, though.

**Comment by nicoguaro** [comment]
It seems to be a [bug to Groups in Zotero](https://bibbase.userecho.com/communities/1/topics/23-only-small-part-of-zotero-group-library-is-displayed#comment-92). I managed to got it to work, and you can see the complete list in [this](https://bibbase.org/show?bib=https://bibbase.org/zotero-group/nicoguaro/525293). The use is

    https://bibbase.org/show?bib=https://bibbase.org/zotero-group/USERID/GROUPID

so, in our case, it would be something like

    https://bibbase.org/show?bib=https://bibbase.org/zotero-group/nicoguaro/525293

Currently, we have 471 entries. I updated the database to include all missing, names, titles, and dates. Nevertheless, I think that we might reorder the whole database a little bit. I know that @asmeurer suggested creating some general groups in the past, should we do something like that?
